qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -272,18 +272,20 @@ public void testAllIndicesWithIndexFilter() throws Exception {
     @SuppressWarnings("unchecked")
     // Returns a client connected to one of the upgraded nodes.
     private RestClient getUpgradedNodeClient() throws IOException {
+        // TODO[lor]: review this after we moved NodeInfo to qualified ids (Build.current().version())
+        var currentVersion = Build.current().version().replace("-SNAPSHOT", "");
         for (HttpHost host : getClusterHosts()) {
             RestClient client = RestClient.builder(host).build();
             Request nodesRequest = new Request("GET", "_nodes/_local/_none");
             Map<String, ?> nodeMap = (Map<String, ?>) entityAsMap(client.performRequest(nodesRequest)).get("nodes");
             Map<String, ?> nameMap = (Map<String, ?>) nodeMap.values().iterator().next();
             String version = (String) nameMap.get("version");
-            if (version.equals(Version.CURRENT.toString())) {
+            if (version.replace("-SNAPSHOT", "").equals(currentVersion)) {
```

Comment:
Do we need the snapshot replacement here? I wouldn't expect the  snapshot qualifier in the version request.

Reply:
After https://github.com/elastic/elasticsearch/pull/100868 the replace is not needed anymore, removed.